src/gcp/devConnect.ts

```diff
@@ -234,6 +234,46 @@ export async function listAllLinkableGitRepositories(
   return repos;
 }
 
+/**
+ * Lists all branches for a given repo. Returns an array of branches
+ * and a map to make lookups more efficient.
+ */
+interface ListAllBranchesResponse {
+  branches: string[];
+  lookupMap: Map<string, boolean>;
```

Comment:
if this is just to help with lookups, how about `Set`? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
makes sense!